### DISCUSSION TITLE
Github login on preview environment?

### DISCUSSION THREAD

**Discussion by AxelBriche** [vercel/community#1792] [Help]
Hi, is it possible to make the github login work on the Vercel preview environment?
I can use github login on my website in production, but not in preview.

**Comment by amyegan** [comment]
Hi @AxelBriche. Can you tell me more about what's going wrong?

If this is about Preview Deployment Protection, I need to know more about your project settings so I can try to reproduce the error.

If you're trying to implement a GitHub login feature on your own site, you may need to double check that your preview environment variables do not have expired or invalid values.

If this is about the Preview Comments feature, the "Log in to comment" link in the preview comments overlay should send you to the normal Vercel log in page with an option to "Continue with GitHub". I just double checked, and it is working correctly for me.

**Comment by AxelBriche** [comment]
I have the "GitHub login feature" on my multi-tenant website and all subdomains.
This feature is and works.

But this feature needs an "Authorization callback URL" and this url changes every time in the Vercel preview environment.
So I can't test the login on the preview environment.

I build a website like Hashnode and I only plan social login and login with magic link.
Test that login is ok in preview environment is important for my project.

![CleanShot 2023-03-16 at 00 16 25@2x](https://user-images.githubusercontent.com/1295251/225465568-3545ddcf-10e0-465d-a563-dd7cf52e9962.png)

**Comment by amyegan** [comment]
You can use a specific branch, sort of like in the [staging environment guide](https://vercel.com/guides/set-up-a-staging-environment-on-vercel), combined with the [git branch generated URL](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) that is automatically assigned to the latest deployment from that branch. That would give you a consistent preview callback URL.
